### ISSUE TITLE
KillMode=none ломает старт при запуске systemd

### ISSUE THREAD

**Issue by Listum** [bol-van/zapret#937]
vyos systemd[1]: /lib/systemd/system/zapret.service:10: Unit uses KillMode=none. This is unsafe, as it disables systemd's process lifecycle management for the service. Please update the service to use a safer KillMode=, such as 'mixed' or 'control-group'. Support for KillMode=none is deprecated and will eventually be removed.
vyos systemd[1]: Stopping zapret.service...

**Comment by Listum** [comment]
Не вник в проблему

**Comment by Listum** [comment]
Даже учитывая то что я поменял KillMode на mixed, мне всё равно приходится перезапускать запрет после запуска vyos

**Comment by bol-van** [comment]
Этот варининг уже лет 10 и ни на что не влияет. systemd вторичен для скриптов запуска zapret, его возможности не используются. Его задача - лишь дернуть init.d/sysv/zapret

Что значит перезапускать после запуска vyos ? Перезапускать предполагает он был запущен, и его надо остановить и запустить снова. подробнее опишите процесс.
Что такое vyos ? Это ОС, контейнер ?
Не работает по какой причине ? Не применены tables, не запущены демоны ?
Может там гонки есть ? Например, встроенный фаервол убивает правила zapret ?
в варианте systemd нет никаких прикруток к фаерволам, при возникновении проблем с этим надо разделять применение фаервола и запуск демонов как описано в доке

вообщем я прекращаю догадываться, а вы описываете детали

**Comment by Listum** [comment]
VyOS это дистрибутив на основе Linux предназначенный для маршрутизации, роли DHCP-сервера, DNS-сервера и т.п. и т.д. Каждый раз при запуске роутера zapret почему-то не хочет работать. 
Моё предположение: Для VyOS нужно время чтобы съесть конфигурацию и после он начинает шаманить с nftables, переписывая правила который zapret вносит сразу при старте демона.

journalctl zapret: 
https://pastebin.com/NUJtRf26

**Comment by Listum** [comment]
>Этот варининг уже лет 10 и ни на что не влияет.
Тогда, возможно, это было просто совпадение.

**Comment by bol-van** [comment]
Что случилось с правилами легко проверить через
nft list table inet zapret
или
iptables-save

Если используются iptables, то лучше с них уйти на nftables, если есть такая возможность.
iptables все в куче, их легко может снести фаервол. в nft вряд ли будут трогать отдельную таблицу

**Comment by bol-van** [comment]
Вижу, что уже на nftables. Хорошо. Тогда стоит проверить таблицу nft zapret и запущены ли процессы nfqws.
В момент проблемы

**Comment by Listum** [comment]
![image](https://github.com/user-attachments/assets/331b5acd-87a1-4d1b-b4c3-9e8f17d738da)
[zapret_table.txt](https://github.com/user-attachments/files/18135314/zapret_table.txt)

**Comment by Listum** [comment]
![image](https://github.com/user-attachments/assets/7c0eb32b-038b-47c2-a1cc-652e58ff3efd)

**Comment by Listum** [comment]
После рестарта в nft добавляется 
`flowtable ft {
                hook ingress priority filter - 1
                devices = { eth4.10 }
        }
`

**Comment by bol-van** [comment]
flow offload включен в config ?
Для flow offload нужны входящие и исходящие интерфейсы. Они должны быть определены а IFACE_WAN и IFACE_WAN
На момент инициализации flowtable они уже должны существовать.
Если юнит стартует до их создания, нужно либо делать какую-то синхронизацию post-network, либо вешать хук и вызывать /opt/zapret/init.d/sysv/zapret start_fw
Или отключить offload

**Comment by Listum** [comment]
Тогда придётся отключить off-load. Оставлю создание хука на потом, как ресурсов мало будет. Спасибо большое за помощь

**Comment by bol-van** [comment]
Даже если offload и не работает, то это не должно влиять на работоспособность.
Надо вставить запись дебаг лога в файл --debug=@/path/to/log.txt
и смотреть что происходит в проблемном состоянии

тк дистриб явно не прошивочный, в чем смысл software offloading ? очень большой трафик и очень слабая железка ?

**Comment by Listum** [comment]
Да. Пока что железяка мощная, но хотелось PnP образ диска для слабой машинки

**Comment by Listum** [comment]
>--debug=@/path/to/log.txt

А куда вставлять?

**Comment by bol-van** [comment]
В параметры nfqws

**Comment by Listum** [comment]
![image](https://github.com/user-attachments/assets/157d27f4-178e-4185-8eab-4115069d2f5a)

**Comment by bol-van** [comment]
Если это все, то на него ничего не приходит.
Либо трафика нет, либо он идет на другой интерфейс. Не тот, что в множестве wanif.

```
	set wanif {
		type ifname
		elements = { "eth4.10" }
	}

```

**Comment by Listum** [comment]
Да. Перезапустил и на него пошёл трафик. Почему же так происходит?

**Comment by Listum** [comment]
> eth4.10

Ну это 100% мой wan интерфейс

**Comment by bol-van** [comment]
сравните в обоих состояниях
`nft list set inet zapret wanif`

в остальном то, что я вижу в таблице, заполнено правильно. должно как бы идти

**Comment by Listum** [comment]
Идентично.

**Comment by Listum** [comment]
Очень интересная ситуация, конечно

**Comment by bol-van** [comment]
Теоретически могут быть правила в других таблицах.
Если это железка специализированная какая-то, там может быть оффлоад

**Comment by Listum** [comment]
Отключено по умолчанию

**Comment by Listum** [comment]
>Теоретически могут быть правила в других таблицах.

https://pastebin.com/ZfSf9C4E

**Comment by Listum** [comment]
Появилась догадка.
Возможно каждый раз VyOS заново пересоздаёт VLAN интерфейсы, а возможно и вообще все. Zapret/nfqws не может забиндиться на интерфейс и в итоге не пропускает через себя трафик

**Comment by bol-van** [comment]
В  параметрах есть --bind-fix ?

**Comment by Listum** [comment]
Нет. Добавить?

**Comment by bol-van** [comment]
Нет. Бинд есть только на этих параметрах, причем только на отсылку.
Само по себе получение данных из очереди никак не завязано на какие-либо интерфеймы

**Comment by Listum** [comment]
Тогда всё становиться ещё более загадочным

**Comment by Listum** [comment]
РЕШЕНИЕ: 
У VyOS, запуск сторонних программ, взаимодействующих с сетью, должен осуществляться через скрипт `/config/scripts/vyos-postconfig-bootup.script`. Для работоспособности zapret требуется отключить сервис через systemctl и запускать сервис через `vyos-postconfig-bootup.script`.

```
systemctl disable zapret
echo 'systemctl start zapret' >> /config/scripts/vyos-postconfig-bootup.script
```
